lib/tasks/plugins.rake

```diff
@@ -25,6 +25,28 @@ namespace :plugin do
       ActiveRecord::Tasks::DatabaseTasks.migrations_paths << path
     end
   end
+
+  task :rubocop, [:engine, :junit] => :environment do |t, args|
+    unless args[:engine]
+      abort("You must specify the name of the plugin (e.g. rake plugin:rubocop['my_plugin'])")
+    end
+
+    engine = "#{args[:engine].tr('-', '_').camelize}::Engine".constantize
+    options = []
+
+    options += ['--format', 'progress']
+    options += ['--format', 'junit', '--out', args[:junit]] if args[:junit]
+
+    config_path = engine.root.join('.rubocop.yml')
+    options += ['--config', config_path.to_s] if config_path.exist?
+
+    options << engine.root.join('{app,lib,test}/**/*.rb').to_s
```

Comment:
Isn't this part of the configuration .rubocop.yml?

Reply:
Doesn't look like it, at least not by default. You would also still need to set the root (or chdir). I guess we can also flip it to not pass patterns, chdir to `engine.root` and pass `--config` with Foreman's .rubocop.yml if it doesn't exist in the plugin root.

Reply:
would it then be better to call the plugin rubocop task (which then set the necessary rubocop options) instead of defining a completely new task?

Reply:
Which task? The main rubocop task has no parameters, right? And what's the benefit of having the indirection of a Rake task?

Reply:
the foreman plugin rubocop task (this PR) executes the plugin rubocop task like https://github.com/theforeman/foreman_bootdisk/blob/36ba85a4651ae89c3021e086762b866077e5e9a8/lib/tasks/bootdisk.rake#L95 so that only the foreman_bootdisk files are tested using rubocop. I don't know if this is possible - just an idea.

Reply:
I looked at that, but decided it wasn't worth anything. If you read https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/rake_task.rb, it's just a wrapper around `RuboCop::CLI` in the form of a Rake task. Defining the task inline and then calling it is just overhead.

Reply:
make sens! thanks.